### PR TITLE
test(ci): introduce Github Actions

### PR DESCRIPTION
**Pull request by sammyne**
### Description

This PR tackle issue #6, which add Github Actions as CI services to run tests~

### CODE REVIEW

File: .github/workflows/sgx.yml [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+    branches: [ 'sgx**' ]
+  pull_request:
+    branches: [ 'sgx**' ]
+
+jobs:
+  test:
+
+    runs-on: ubuntu-18.04
+    container: 
+      image: baiduxlab/sgx-rust:1804-1.1.1
```
Comment (mssun) [inline .github/workflows/sgx.yml]: Can we use this image `teaclave/teaclave-build-ubuntu-1804-sgx-2.9:0.1.0`? Thanks.

File: .github/workflows/sgx.yml [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    - name: Cache cargo index
+      uses: actions/cache@v1
+      with:
+        path: ~/.cargo/git
+        key: ${{ runner.os }}-cargo-index-${{ hashFiles('**/Cargo.toml') }}
+        restore-keys: ${{ runner.os }}-cargo-index- 
+    - name: Run tests
+      run: |
+        . ~/.cargo/env && 
+        . /opt/sgxsdk/environment &&
```
Comment (mssun) [inline .github/workflows/sgx.yml:47]: I believe this is not needed if using this `teaclave/teaclave-build-ubuntu-1804-sgx-2.9:0.1.0` image.

File: .github/workflows/sgx.yml (hunk 1 above)
Comment (sammyne) [reply .github/workflows/sgx.yml]: No problem

File: .github/workflows/sgx.yml (hunk 2 above)
Comment (sammyne) [reply .github/workflows/sgx.yml:47]: I think not. This line addressed the hardcoded HOME (/home/github) directory set by Github Actions. @mssun

**Pull request merged**